app/errors.py

```diff
@@ -0,0 +1,20 @@
+class VaccineError(Exception):
+    pass
+
+
+class NotVaccinatedError(VaccineError):
+    def __init__(self) -> None:
+        self.message = "The visitor does not have a vaccine."
+        super().__init__(self.message)
+
+
+class OutdatedVaccineError(VaccineError):
+    def __init__(self) -> None:
+        self.message = "The vaccine must not be expired."
```

Comment:
You don't really need this row. You can just put the text of message directly to super init.

Reply:
It's a common practice to store the exception message as an instance variable.
 You can see this by following the link attached to the first lesson in the paragraph ["Customizing Exception Classes"](https://www.programiz.com/python-programming/user-defined-exception#class-customize)

Reply:
OK, let it be. But I don't really see big sense in this line.